README.md

# Travel Wallet

Travel Wallet helps you keep manage your loyalty programs and credit cards. It's a free app packed with useful features for travelers.

Features:

- Specifically designed for travelers. No internet connection required.
- 100% Ad-free
- Database of 300+ credit cards from US and Canada
- Database of 50+ loyalty programs (airlines, hotels, rental cars, etc.)
- Supports multiple users
- Calculates Chase 5/24 status (US Only)

## Installation

This application is listed on the Google Play store and can be downloaded and installed directly from there.

This application requires Android 4.0.3 "Ice Cream Sandwich" or higher (API 15+).

[Google Play Store Link](https://play.google.com/store/apps/details?id=com.davidlcassidy.travelwallet "Travel Wallet")

## Getting Started

### Introduction

Travel Wallet primarily consist of three interfaces, each with its own tab and display window.

- Notifications Interface - Displays notifications from both the loyalty programs and the credit cards
- Loyalty Program Interface - Provides users the ability add, view, edit, and delete loyalty programs
- Credit Card Interface - Provides users the ability add, view, edit, and delete credit cards

Along with these three interfaces, there are two buttons located on the top of the main screen. The first is the Users button allowing the user to manage multiple program/card users. The second is a dropdown menu where the user can modify app settings, open a summary report, view the developer's note, and upgrade to Travel Wallet Pro if they have not already.

### Notifications Interface

Both loyalty program and credit card notifications will appear in the first tab titled Notifications and are sorted by time remaining until the approaching event date. There are currently two types of event notifications that are sent:

- Upcoming loyalty program points expiration
- Upcoming credit card annual fee

In the Settings, the user can adjust the period of time before these events that a notification will be sent. Furthermore, monitoring of each individual loyalty program and credit card can be turned on or off, dictating whether or not notifications will be sent. This makes for a flexible notifications interface that is customizable to the needs of each user.

Along with the Notifications tab, Travel Wallet can also send push notifications to the phone status bar in real time. This can be useful for users who would like to be automatically notified of events rather than regularly opening the app and manually checking their notifications. These phone notifications can be turned on/off in the Settings. 

Phone notifications can be cleared, but individual notifications within the app can not be cleared or snoozed. They will remain until they are resolved or their monitoring turned off. Finally, note that clicking on each notification within the app will route the user to the loyalty program or credit card causing the notification.

### Loyalty Programs Interface

Loyalty programs will appear in the second tab titled Loyalty Programs. New loyalty programs are added by clicking the round plus button in the bottom right corner. Once programs have been added, the main screen will show a list of all added programs. The sort order and primary display field can be configured in the Settings. Clicking on a program will pull up that program's detail screen displaying all fields for that program along a button to toggle the notification monitoring for the specific program. On the top of the detail screen, there are also buttons for editing or deleting the specific program.

### Credit Cards Interface

Credit cards will appear in the third tab titled Credit Cards. The credit card interface functions essentially the same as the loyalty program interface in regards to adding, viewing, editing, and deleting credit cards.

### Travel Wallet Pro

The free version of Travel Wallet is limited to two users, ten loyalty programs, and ten credit cards. Purchasing Travel Wallet Pro will release these limits and unlock an exclusive Gold color scheme. There is an option to upgrade to Travel Wallet Pro in the dropdown menu in the top right corner of the main screen.

## Screenshots

<p align="center">
  <img src="screenshots/basic/screenshot_1.png" width="250">
  <img src="screenshots/basic/screenshot_2.png" width="250">
  <img src="screenshots/basic/screenshot_3.png" width="250">
</p>

<p align="center">
  <img src="screenshots/basic/screenshot_4.png" width="250">
  <img src="screenshots/basic/screenshot_5.png" width="250">
  <img src="screenshots/basic/screenshot_6.png" width="250">
</p>

## Building from Source

### Requirements

- Android SDK v15+
- Latest Android Build Tools
- Android Support Repository

### Installation

This project uses the Gradle build system.

1. Download the project by cloning this repository or downloading a snapshot.
1. In Android Studio, create a new project and choose the "Import Project" option.
1. Select the "Travel Wallet" directory that you downloaded with this repository.
1. If prompted for a gradle configuration, accept the default settings. Alternatively use the "gradlew build" command to build the project directly.

## Contributions

Pull requests and GitHub issues are welcome!

If you would like to do a pull request, please get in touch with me before you start writing code so we 
can avoid duplicated effort or unnecessary work.

## Copyright

Travel Wallet Android App

Copyright (C) 2021 David L Cassidy. All rights reserved.


CHANGELOG.md
v1.4.10 : 2021-04-28
Feature Release

 - Renamed Owners to Users
 - Added additional anti-churning rules
 - Added support for regional number patterns
 - Updated reference database

v1.4.9 : 2019-07-25
Feature Release

 - Updated reference database

v1.4.8 : 2018-12-21
Feature Release

 - Added "Other" country option

v1.4.7 : 2018-11-07
Feature Release

 - Updated reference database

v1.4.6 : 2018-11-05
Patch Release

 - Fixed bug Chase 5/24 bug counting foreign cards
 - Fixed add/edit card credit limits in foreign currencies

v1.4.5 : 2018-10-25
Feature Release

 - Reduced size of logos/icons (reduced size of app by 18%)
 - Added support for second country (Canada)
 - Created custom single choice dialog layouts
 - Updated reference database

v1.4.4 : 2018-10-17
Feature Release

 - Updated PurchasePro description

v1.4.3 : 2018-10-17
Patch Release

 - Fixed summary next date calculations
 - Fixed bug with notification month plurals ("1 months")
 - Updated screenshots
 

v1.4.2 : 2018-10-17
Feature Release

 - Added color scheme setting
 - Updated IAB for dynamic pricing
 - Updated developer's notes
 - Updated reference database

v1.4.1 : 2018-10-12
Patch Release

 - Fixed IAB bug, due to incorrect public key

v1.4.0 : 2018-10-11
Minor Release

 - Added fully-featured pro app version
 - Added In-app Billing ability

v1.3.3 : 2018-07-22
Feature Release

 - Fixed bug with program type filter
 - Added ability to toggle filters on/off in settings
 - Updated notification button to change color when clicked
 - Reorganized settings items

v1.3.2 : 2018-06-29
Feature Release

 - Updated database indexing and cursors
 - Added 13 new credit cards as requested by users

v1.3.1 : 2018-06-25
Feature Release

 - Updated owners detail activity layout
 - Updated summary report (minor)

v1.3.0 : 2018-06-24
Minor Release

 - Added owners interface allowing for multiple program/card owners
 - Added Chase 5/24 status support
 - Added program/card filters
 - Added support for custom program/card notes
 
v1.2 : 2018-06-15
Patch Release

 - Fixed bug with popup date picker dialog causing stability issues
 
v1.1 : 2018-06-14
Patch Release

 - Fixed bug with the reference database creation causing stability issues
 
v1.0 : 2018-06-09
Major Release

 - Initial release

app/src/main/java/com/davidlcassidy/travelwallet/Enums/ItemField.java
/*
 * Travel Wallet Android App
 * Copyright (C) 2021 David L Cassidy. All rights reserved.
 * Last modified 4/28/21 11:39 AM
 */

package com.davidlcassidy.travelwallet.Enums;

/*
ID Local Cache : AppPreferences
WARNING : Changing ID values may change app settings to the default value within current app installs.
 */

public enum ItemField {

    // Enum members
    TYPE(11, ItemType.LOYALTY_PROGRAM, "Type"),
    PROGRAM_NAME(12, ItemType.LOYALTY_PROGRAM, "Program Name"),
    ACCOUNT_NUMBER(13, ItemType.LOYALTY_PROGRAM, "Account Number"),
    POINTS(14, ItemType.LOYALTY_PROGRAM, "Points"),
    VALUE(15, ItemType.LOYALTY_PROGRAM, "Value"),
    EXPIRATION_DATE(16, ItemType.LOYALTY_PROGRAM, "Expiration Date"),
    PROGRAM_NOTES(17, ItemType.LOYALTY_PROGRAM, "Program Notes"),

    BANK(21, ItemType.CREDIT_CARD, "Bank"),
    CARD_NAME(22, ItemType.CREDIT_CARD, "Card Name"),
    OPEN_DATE(23, ItemType.CREDIT_CARD, "Open Date"),
    AF_DATE(24, ItemType.CREDIT_CARD, "Annual Fee Date"),
    ANNUAL_FEE(25, ItemType.CREDIT_CARD, "Annual Fee"),
    CARD_NOTES(26, ItemType.CREDIT_CARD, "Card Notes"),

    USER_NAME(31, ItemType.USER, "User Name"),
    ITEM_COUNTS(32, ItemType.USER, "Item Counts"),
    PROGRAMS_VALUE(33, ItemType.USER, "Programs Value"),
    CREDIT_LIMIT(34, ItemType.USER, "Credit Limit"),
    USER_NOTES(35, ItemType.USER, "User Notes");

    private final int id;
    private final ItemType itemType;
    private final String name;


    ItemField(int id, ItemType itemType, String name) {
        this.id = id;
        this.itemType = itemType;
        this.name = name;
    }

    // Returns item status from ID
    public static ItemField fromId(int id) {
        for (ItemField itemField : ItemField.values()) {
            if (itemField.getId() == id) {
                return itemField;
            }
        }
        return null;
    }

    // Returns item field from name
    public static ItemField fromName(String name) {
        for (ItemField itemField : ItemField.values()) {
            if (itemField.getName().equals(name)) {
                return itemField;
            }
        }
        return null;
    }

    // Returns item field ID
    public int getId() {
        return id;
    }

    // Returns item field name
    public String getName() {
        return name;
    }
}